lib/std/spinlock.zig

```diff
@@ -20,9 +22,56 @@ pub const SpinLock = struct {
     }
 
     pub fn acquire(self: *SpinLock) Held {
-        while (@atomicRmw(u8, &self.lock, builtin.AtomicRmwOp.Xchg, 1, AtomicOrder.SeqCst) != 0) {}
+        var backoff = Backoff.init();
+        while (@atomicRmw(u8, &self.lock, .Xchg, 1, .Acquire) != 0)
+            backoff.yield();
         return Held{ .spinlock = self };
     }
+
+    pub fn yieldCpu() void {
+        switch (builtin.arch) {
+            .i386, .x86_64 => asm volatile("pause" ::: "memory"),
```

Comment:
One more thing-

I doubt it matters much, but is this memory clobber correct? I don't think `pause` instruction can modify memory. The atomic loads and stores elsewhere in the code already communicate to the compiler that the memory they address can be modified by external threads. I think this is an unnecessary clobber.

Reply:
Most likely not, it was lifted [directly out of mimalloc](https://github.com/microsoft/mimalloc/blob/master/include/mimalloc-atomic.h#L214) without much thought put into it. Also reminds me about the memory ordering on [these](https://github.com/ziglang/zig/blob/92dac89d019fcd91aa043a116d6a6e166da8c4b2/lib/std/parker.zig#L160) [two](https://github.com/ziglang/zig/blob/92dac89d019fcd91aa043a116d6a6e166da8c4b2/lib/std/parker.zig#L172) atomic loads. Could they be reduced to `Monotonic` given the cmpxchg/fetch_add following them are the only ones that should to be synchronized as producer/consumer?

Reply:
My understanding of park() and unpark() is that they do not protect resources; they only deal with putting a thread to sleep getting woken up. In this case Monotonic is fine, because moving non-atomic memory loads and stores before or after park() or unpark() would be OK.